.github/workflows/ci.yml

```diff
@@ -0,0 +1,156 @@
+# CI WorkFlow for build, test and release of Plugin Packages
+name: CI for Plugin Packages
+on:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - 'v*.*.*'
+    paths-ignore:
+      - 'plugins/**'
+      - 'hack/test-plugin-manifests.sh'
+      - 'hack/update-plugin-manifests.sh'
+      - 'hack/update-preflight-manifest.sh'
+      - 'hack/update-log-collector-manifest.sh'
+      - '.github/workflows/plugins.yml'
+  pull_request:
+    branches:
+      - '*'
+    paths-ignore:
+      - 'plugins/**'
+      - 'hack/test-plugin-manifests.sh'
+      - 'hack/update-plugin-manifests.sh'
+      - 'hack/update-preflight-manifest.sh'
+      - 'hack/update-log-collector-manifest.sh'
+      - '.github/workflows/plugins.yml'
+
+jobs:
+  pre-build:
+    runs-on: ubuntu-latest
+    outputs:
+      preflight: ${{ steps.filter.outputs.preflight }}
+      logCollector: ${{ steps.filter.outputs.logCollector }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Run yamllint
+        uses: reviewdog/action-yamllint@v1
+        with:
+          fail_on_error: true
+          reporter: local
+          filter_mode: nofilter
+
+      - name: Run golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: v1.29
+
+      - name: Run Shell lint
+        run: hack/run-shell-lint.sh
+
+      - name: Verify code patterns
+        run: hack/verify-code-patterns.sh
+
+      - name: Detect File Changes
+        uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          base: ${{ github.ref }}
+          filters: |
+            preflight:
+              - 'hack/**'
+              - '.krew/tvk-preflight.yaml'
+              - 'tools/preflight/**'
+              - '.goreleaser.yml'
+            logCollector:
+              - 'cmd/**'
+              - 'hack/**'
+              - '.krew/tvk-log-collector.yaml'
+              - 'tools/log_collector/**'
+              - '.goreleaser.yml'
+
+  preflight:
+    runs-on: ubuntu-latest
+    needs: pre-build
+    if: (needs.pre-build.outputs.preflight == 'true') || startsWith(github.ref, 'refs/tags/')
```

Comment:
add a comment as to why we need both of these conditions

Reply:
yes.. done. Please review